### DISCUSSION TITLE
Add AI assistant plugin like on Krita to aid drawing or brush by text or add effects using words

### DISCUSSION THREAD

**Discussion by ForestEco** [opentoonz/opentoonz#6363] [Ideas]
### Is your feature request related to a problem? Please describe

Check out this interesting tool in the video that renders at a very customized level AI by inpainting etc rather than generate the whole thing, and collage their designs with expert quality to the desired extent. It would speed up design and you can simply reference a style or coloring. You don't have to copy exactly anything from another artist at all. I could see artists layering their work easily and enhancing their designs with less effort. Think of it like applying effects to the image using text and your pen stroke selection rather than a limited number of effects from the drop down. It also just copies your own hand movements to smooth lines, etc. 
Ultimately the artist will finish the design.

https://github.com/Acly/krita-ai-diffusion
https://youtu.be/AU8NDSBIS1U?si=H7qKit13g_9-9aPF

### Describe the solution you'd like

Try embedding the plugin used here in OpenToonz after getting permission: https://github.com/Acly/krita-ai-diffusion

![Uploading 2024-03-11_23-22-32.png…]()

Also is there a plugin the add a lasso tool that allows you to use the paint n fill from the selection because right now I have to toggle to a color and free hand doesn't allow fill.

**Comment by ForestEco** [comment]
All valid points, as I am finding that there's not much unity to AI rendering after all for scenes in general, your words are moved by more than opinion.

For AI, I was aiming to use it more like a stencil, or search engine for texture/styles to try and get some unity to some designs or collage something. That way I can color over the lines I draw that are a bit improved, but some people like that rough sketch look more too. However you're right, Krita may be useful enough. But I am looking for the bone rendering tools to make it easier to animate that Krita doesn't have and I find OT a bit easier for animation than it.

**Comment by RodneyBaker** [comment]
This might be a good time to point out that some are leaning into 'AI' is various ways.
@konero, for instance, has a GPT app for assisting with creating/adapting shaders for use with Opentoonz.
I've created a few GPT apps to test the waters, one of which is the only GPT app that shows up right now if 'Opentoonz' or 'Tahoma2D' is searched for in ChatGPT.  Link: https://chat.openai.com/g/g-DmHRDKGTj-opentoonz-tech-advisor

That Tech Advisor doesn't currently have much specific knowledge of rigging for opentoonz so it won't be much help there.
It is very limited also in that it requires access to ChatGPT4 etc. as GPT apps are not available via the free access at this point.
The OT Tech Advisor's main knowledge base is just the Opentoonz and Tahoma2D documentation although it also has some experimental data that allows it to create custom panels so that Qt Designer doesn't need to be used for that.  It also has some example mypaint brushes.

Will Opentoonz gain AI capabilities?  I'm pretty sure it will although it remains to be seen in what ways that might occur.
I do hope it's not really as a plugin as plugins aren't always ideal.  
My take is that just bringing python into the mix when dealing with text based files is a good initial approach to take and with python being a core programming language used with AI models that alone will open many opportunities.

Can Opentoonz add more enroads to leverage AI assistants?  Certainly.
The Opentoonz ecosystem was actually one of the earliest to lean into machine learning via plugins... although not with user ease of use as one of its strong points.  Look into the Neural Style plugin for more on that.

You added to your post:
> Also is there a plugin the add a lasso tool that allows you to use the paint n fill from the selection because right now I have to toggle to a color and free hand doesn't allow fill.  

Opentoonz actually has a feature that may address this need.
The Fill Tool has an option 'Pick+Freehand' that allows for lassoing areas which then transfers color fill into areas.

![image](https://github.com/opentoonz/opentoonz/assets/2576432/d4ac7729-6841-47f6-8cab-7f5f1feb76c8)

**Comment by ForestEco** [comment]
Thanks for the input. I know how to do some AI development myself but not a whole lot of the deeper frameworks. 
I would use a StableDiffusion model and feed it through Comfyui to get on the fly edits. Also LMStudio does host AI models offline so no need for Chatgpt. It will be slower because the best GPUs tend to be remote clouds.

As for the lasso tool from fill, I tried it but it seems to apply color to only existing lines or areas that were drawn prior and not allow you to draw with the lasso tool itself on blank canvas without penned drawings, so using it to create shapes on the canvas and not modify the existing work. 

Maybe I am missing an intermediate step using the fill tool to do it? I tried a few tutorials but nothing yet.
